### DISCUSSION TITLE
Adding Characters to font spritesheets

### DISCUSSION THREAD

**Discussion by CCShakeCat** [Techokami/SonicWorldsNext#216] [Q&A] [answered]
I am making a Sonic fan game with this engine and I am modifying the pause texts. The font file included does not have a Question Mark (**?**) or an Exclamation Mark (**!**). How do I add characters for the texts?

1. 
![image](https://github.com/user-attachments/assets/b1e6e097-19e9-45fa-bcf1-0dc015517581)
2. 
![image](https://github.com/user-attachments/assets/bfefa4d2-9ae9-4063-ac85-c64a240644fe)

**Comment by r6915ee** [comment, marked as answer]
You are dealing with a native Godot bitmap font here. These are a very specific format in which the import of an image defines how it acts as a font. They usually must match the ASCII range.

In your case, you need to reimport the file and append these two items to the character ranges parameter:

`33`
`63`

**Reply by CCShakeCat** [reply]
That worked! Thank you!
